File: templates/countANNOTSVfeatures.R
args <- commandArgs(trailingOnly = TRUE)
lapply(c("data.table","dplyr","ggplot2","tidyr"), require, character.only = TRUE)
          
input_file <- NULL
for (i in seq_along(args)) {
    if (args[i] == "--input") {input_file <- args[i + 1]}
    else if (args[i] == "--interval") {interval <- args[i + 1]}
    else if (args[i] == "--sample") {sample <- args[i + 1]}
    else if (args[i] == "--original_file_name") {original_file_name <- args[i + 1]}
}
# Read input file using fread
if (!is.null(input_file)) {
    data <- fread(input_file, h=TRUE)
} else {stop('VEP annotated input file not specified')}
if (is.null(interval)) { interval <- "all" } 
if (is.null(sample)) { sample <- "all" }  


annotsv <- data %>% 
    filter(!duplicated(AnnotSV_ID)) %>%
    select(AnnotSV_ID,SV_type,GnomAD_pLI,B_loss_source,B_gain_source) %>% 
    mutate(novel_annotsv = ifelse(is.na(GnomAD_pLI) & 
        B_loss_source=="" & 
        B_gain_source=="",
        "TRUE", "FALSE")) %>%
    group_by(novel_annotsv,SV_type) %>%
    count(SV_type, name = "Count") %>% 
    select(var=SV_type,count=Count,novel_annotsv) %>%
    mutate(interval = interval,
        sample = sample,
        original_file_name = original_file_name) %>%
    fwrite(sprintf("%s.%s.annotsv.counted", interval, sample), col.names=FALSE, sep="\t")

File: templates/countANNOTSVfeatures.py
import sys
import pandas as pd

# Check if the required modules are installed
required_modules = ['data.table', 'dplyr', 'ggplot2', 'tidyr']
for module in required_modules:
    try:
        __import__(module)
    except ImportError:
        print(f"Module {module} not found")
        sys.exit(1)

input_file = None
interval = "all"
sample = "all"
original_file_name = None

# Parse command line arguments
for i in range(1, len(sys.argv), 2):
    if sys.argv[i] == "--input":
        input_file = sys.argv[i+1]
    elif sys.argv[i] == "--interval":
        interval = sys.argv[i+1]
    elif sys.argv[i] == "--sample":
        sample = sys.argv[i+1]
    elif sys.argv[i] == "--original_file_name":
        original_file_name = sys.argv[i+1]

# Read input file using pandas
if input_file:
    data = pd.read_csv(input_file, sep='\t', header=0)
else:
    print('VEP annotated input file not specified')
    sys.exit(1)

annotsv = data[~data.duplicated(subset='AnnotSV_ID')][['AnnotSV_ID', 'SV_type', 'GnomAD_pLI', 'B_loss_source', 'B_gain_source']]
annotsv['novel_annotsv'] = annotsv.apply(lambda row: True if pd.isna(row['GnomAD_pLI']) and row['B_loss_source']=='' and row['B_gain_source']=='' else False, axis=1)
annotsv = annotsv.groupby(['novel_annotsv', 'SV_type']).size().reset_index(name='Count')
annotsv = annotsv[['SV_type', 'Count', 'novel_annotsv']]
annotsv['interval'] = interval
annotsv['sample'] = sample
annotsv['original_file_name'] = original_file_name

# Write the output to a file
output_file = f"{interval}.{sample}.annotsv.counted"
annotsv.to_csv(output_file, sep='\t', index=False)
